src/DssTest.sol

```diff
@@ -323,6 +325,106 @@ abstract contract DssTest is Test {
         checkFileAddress(_base, _contractName, values);
     }
 
+    /// @dev This is forge-only due to event checking
+    function checkFileString(address _base, string memory _contractName, string[] memory _values) internal {
+        FileLike base = FileLike(_base);
+        uint256 ward = base.wards(address(this));
+
+        // Ensure we have admin access
+        GodMode.setWard(_base, address(this), 1);
+
+        // First check an invalid value
+        vm.expectRevert(abi.encodePacked(_contractName, "/file-unrecognized-param"));
+        base.file("an invalid value", "");
+
+        // Next check each value is valid and updates the target storage slot
+        for (uint256 i = 0; i < _values.length; i++) {
+            string memory value = _values[i];
+            bytes32 valueB32;
+            assembly {
+                valueB32 := mload(add(value, 32))
+            }
+
+            // Read original value
+            (bool success, bytes memory result) = _base.call(
+                abi.encodeWithSignature(string(abi.encodePacked(value, "()")))
+            );
+            assertTrue(success);
+            string memory origData = abi.decode(result, (string));
+            string memory newData;
+            newData = string.concat(newData, " - NEW");
```

Comment:
Any reason to not do directly?
```suggestion
            string memory newData = "NEW";
```

Reply:
Oh, I guess I copied from the `checkFileUint` structure, and it had an `unchecked {}` block, so it stayed like this.